cmd/juju/application/deploy.go

```diff
@@ -387,80 +387,85 @@ type DeployCommand struct {
-  mediawiki-single
-  bundle/mediawiki-single
+A versioned charm URL will be expanded as expected. For example, 'mysql-33'
+becomes 'cs:xenial/mysql-33'.
 
-The current series for charms is determined first by the 'default-series' model
-setting, followed by the preferred series for the charm in the charm store.
+A bundle can be expressed similarly, but not by series:
 
-In these cases, a versioned charm URL will be expanded as expected (for
-example, mysql-33 becomes cs:precise/mysql-33).
+  juju deploy mediawiki-single
+  juju deploy bundle/mediawiki-single
+  juju deploy cs:bundle/mediawiki-single
 
-Charms may also be deployed from a user specified path. In this case, the path
-to the charm is specified along with an optional series.
+A local charm may be deployed by specifying the path to its directory:
 
-  juju deploy /path/to/charm --series trusty
+  juju deploy /path/to/charm --series xenial
 
-If '--series' is not specified, the charm's default series is used. The default
-series for a charm is the first one specified in the charm metadata. If the
-specified series is not supported by the charm, this results in an error,
-unless '--force' is used.
+A local bundle may be deployed by specifying the path to its YAML file:
 
-  juju deploy /path/to/charm --series wily --force
+  juju deploy /path/to/bundle.yaml
 
-Charms can also utilise LXD Profiles when deploying a charm. LXD Profiles can
-be used to override configurations or devices when creating the containers for
-LXD. LXD Profiles are validated against a allow/deny list, using '--force' flag
-can bypass the validation.
+A series is determined in this order of precedence (most preferred to least):
 
-Using the '--force' flag for LXD Profiles is not generally recommended when
-deploying an application; overriding profiles on the container may cause
-unexpected behavior.
+ 1. the '--series' deploy command option
+ 2. the series stated in the charm URL
+ 3. for a bundle, the 'series' field in the charm spec
+ 4. for a bundle, the 'series' field at top level of bundle file
+ 5. the 'default-series' model key
+ 6. the top-most series specified in the charm's metadata file
+    (this sets the charm's 'preferred series' in the Charm Store)
+ 
+If the determined series is not supported by the charm an error is emitted. The
+'--force' option can be used to override this check:
 
-Local bundles are specified with a direct path to a bundle.yaml file.
-For example:
+  juju deploy charm --series xenial --force
 
-  juju deploy /path/to/bundle/openstack/bundle.yaml
+When charms that include LXD profiles are deployed the profiles are validated
+using a hardcoded allow/deny list. This check can be bypassed with the use of
+the '--force' option. However, doing so is not recommended as it can lead to
+unexpected behaviour.
+
+If an 'application name' is not provided, the application name used is the charm
+or bundle name. A user-supplied 'application name' must consist only of
```

Comment:
`or bundle name.`

That isn't quite right.  When deploying a bundle, the `application name` is ignored afaict.

Reply:
:heavy_check_mark: